Automattic/wp-calypso
# Add error-specific validation message in site identification page

Related to https://github.com/Automattic/wp-calypso/issues/91479

## Proposed Changes

Added error-specific validation messages for the following scenarios.

1. Input is an Email
2. Input has invalid characters
3. Wrong Protocol (FTP://, file:// etc.)
4. Missing TLD
5. IDN URLs (Either has non ASCII characters in URL or hostname starts with xn--)
6. General error message for fallback

I have replaced "Incomplete URL Error" from the issue with Wrong Protocol because otherwise the error is too ambiguous and broad to detect. But I am a bit unsure about using the word "Protocol" in the error message, looks too technical. I only kept it because we show it only when the user has already used the wrong protocol and it conveys the exact reason. But LMK if you think a different and less technical message should be shown.

Also, detecting missing TLD is super tricky. For example, "test.com" is valid but "test.wpcomstaging" is not even though both of them has the same structure as a string. So it's not too correct when implemented with regex or some other way through code. But it should be helpful because our error log says users mostly write URLs like this "myblog" instead of "myblog.com" in which case, it'll be helpful.

Also, I've included an example of a valid URL with each error message to better convey to the user how a URL looks.

## Why are these changes being made?
To enable the user to better understand what's wrong with the URL they have input.

## Screenshots

![Screenshot 2024-06-18 at 7 58 38 AM](https://github.com/Automattic/wp-calypso/assets/1190420/f5011e43-b26b-47ee-8583-92f4ee430eb3)

![Screenshot 2024-06-18 at 7 59 14 AM](https://github.com/Automattic/wp-calypso/assets/1190420/89ac6da2-365d-4c79-8edc-a630d0645be3)

![Screenshot 2024-06-18 at 7 59 38 AM](https://github.com/Automattic/wp-calypso/assets/1190420/1388b421-64d0-47df-ab25-e338b9a09442)

![Screenshot 2024-06-18 at 8 00 05 AM](https://github.com/Automattic/wp-calypso/assets/1190420/52d4aadf-9d5e-492c-a0fc-dc4f6dfb9fc9)

![Screenshot 2024-06-18 at 8 00 32 AM](https://github.com/Automattic/wp-calypso/assets/1190420/e0612f33-b7a3-45b3-b21c-faf246908fd4)

![Screenshot 2024-06-18 at 8 01 10 AM](https://github.com/Automattic/wp-calypso/assets/1190420/7a0179c7-31a5-4c12-9e19-8d3ed34accfe)

## Testing Instructions



- Make sure your language is English in https://wordpress.com/me/account
- In the migration flow (take any migration path, for example http://calypso.localhost:3000/start), navigate to the Site Identification step
- In the URL field, try different wrong URLs (Please check the **Proposed Changes** section above). For example - myblog, https://myblog, https://myblog^, email@myblog.com etc.
- You should see different error messages depending on the URL
- Change the language to some other language in `https://wordpress.com/me/account`
- Go to the site identification page and try different wrong URLs again
- Make sure you see the old error message

https://github.com/Automattic/wp-calypso/assets/6820724/4d46c197-2ac1-4ff6-98e7-d80b5034da9c

## Pre-merge Checklist
